csharp/src/iceboxnet/ServiceManagerI.cs

```diff
@@ -814,13 +711,19 @@ private enum ServiceStatus
             Started
         }
 
-        private struct ServiceInfo
+        private class ServiceInfo
         {
-            public string name;
-            public Service service;
-            public Ice.Communicator communicator;
-            public ServiceStatus status;
-            public string[] args;
+            public ServiceInfo(string name, ServiceStatus status, string[] args)
```

Comment:
I assume the ctor and properties should be internal, not public?

Reply:
Yes we can make this internal